### PR TITLE
Migrate peer cache to blockchain data store - Closes #135

### LINKED ISSUE #135: Migrate peer cache to blockchain data store
### Description

Move peer search logic from controller to the blockchain data store. Keep the original structure of the way the data is retrieved. All peers need to be stored in-memory. Distinguish state and stateName.

### Motivation

- Storing data in the local database has a positive impact on performance
- It makes it easier to index and search data
- Proper code organisation makes it possible to use blockchain abstraction layer independently from Lisk Core version underneath

### Acceptance Criteria

- Peer data is available trough the same logic it is done before
- No need for a persistent PouchDB migration
- state and stateName are implemented internally
- The search logic is moved from the controller to the blockchain data store

### PR DESCRIPTION
**Pull request by sameersubudhi**
### What was the problem?
This PR resolves #135 

### How was it solved?
- [x] Peer data is available through the same logic it is done before
- [x] The search logic is moved from the controller to the blockchain datastore
- [x] state and stateName are implemented internally
- [x] Data need not be migrated to PouchDB for persistence

### How was it tested?
Local + Jenkins

### CODE REVIEW

**Review by nagdahimanshu — APPROVED** [review]

File: services/core/methods/controllers/peers.js [hunk 1]
```diff
@@ -99,18 +34,22 @@ const getConnectedPeers = async params => {
 };
 
 const getDisconnectedPeers = async params => {
-	const response = await getPeers(Object.assign(params, { state: 'disconnected' }));
+	const response = await coreService.getPeers(params);
```
Comment (MichalTuleja) [inline services/core/methods/controllers/peers.js:37]: Should there be any constraint?

File: services/core/shared/core/compat/sdk_v3/mappings.js [hunk 2]
```diff
@@ -15,6 +15,13 @@
 //  */
 // const { getDelegateRankByUsername } = require('./delegateCache');
 
+// const reverseMap = (originalMap) => {
+//     const result = {};
+//     Object.entries(originalMap).forEach(([k, v]) => result[v] = String(k).toLowerCase());
+
+//     return result;
+// };
+
```
Comment (MichalTuleja) [inline services/core/shared/core/compat/sdk_v3/mappings.js:24]: Do we still need those lines?

File: services/gateway/sources/mappings/peer.js [hunk 3]
```diff
@@ -20,6 +20,7 @@ module.exports = {
 	os: '=,string',
 	version: '=,string',
 	state: '=,number',
+	stateName: '=,string',
```
Comment (MichalTuleja) [inline services/gateway/sources/mappings/peer.js:23]: @sameersubudhi @nagdahimanshu? `stateName` or `stateText`? What do you think?

**Review by MichalTuleja — CHANGES_REQUESTED** [review]

File: services/core/methods/controllers/peers.js (hunk 1 above)
Comment (sameersubudhi) [reply services/core/methods/controllers/peers.js]: Fixed the typo!

File: services/core/shared/core/compat/sdk_v3/mappings.js (hunk 2 above)
Comment (sameersubudhi) [reply services/core/shared/core/compat/sdk_v3/mappings.js:24]: Yes!

File: services/gateway/sources/mappings/peer.js (hunk 3 above)
Comment (sameersubudhi) [reply services/gateway/sources/mappings/peer.js:23]: I prefer `stateName` for now.
If it weren't for the backward compatibility, I would prefer something like `connectionState` (text) and `connectionStateCode` (number).

Comment (MichalTuleja) [reply services/gateway/sources/mappings/peer.js:23]: Right, I'll keep this in mind for the next API design.

**Review by MichalTuleja — APPROVED** [review]

**Pull request merged**